### PR TITLE
Improved tf.image.adjust_gamma for uint8 images

### PR DESCRIPTION
**Pull request by zimmerrol**
At the moment there is an inconsistency in the `adjust_*` functions of [`tf.image`](https://www.tensorflow.org/api_docs/python/tf/image): While almost all of them work perfectly fine on either `float32` or `uint8` versions of the images by converting them internally at first to the `float32` representation and them applying the modification, the `adjust_gamma` does not behave in the same way. Instead of converting the image at first to `float32` version it immediately applies the transformation. This inconsistency might confuse users or cause an unwanted behavior.

**Changes**:

- inserted a type cast to `float32` and back to the original type like in the other `adjust_*` functions
- updated docstring
- extended unit tests

### CODE REVIEW

**Comment by FlashTek** [comment]
Are you still available to review this PR @drpngx or should someone else take over?

File: tensorflow/python/ops/image_ops_test.py [hunk 1]
```diff
@@ -239,26 +239,31 @@ def testShapeInference(self):
...
       x = constant_op.constant(x_np, shape=x_np.shape)
-      y = image_ops.adjust_gamma(x, gamma=1)
 
-      y_tf = self.evaluate(y)
-      y_np = x_np
+      err_msg = "Gamma should be a non-negative real number."
 
-      self.assertAllClose(y_tf, y_np, 1e-6)
+      try:
+        image_ops.adjust_gamma(x, gamma=-1)
```
Comment (alextp) [inline tensorflow/python/ops/image_ops_test.py]: Use self.assertRaises

**Review by alextp — CHANGES_REQUESTED** [review]

File: tensorflow/python/ops/image_ops_test.py (hunk 1 above)
Comment (FlashTek) [reply tensorflow/python/ops/image_ops_test.py]: Fixed in commit [57835a8](https://github.com/tensorflow/tensorflow/pull/26508/commits/57835a8b9e9a19a061974d1fd393beeaf7898261)

**Review by alextp — DISMISSED** [review]

**Review by alextp — APPROVED** [review]

**Comment by FlashTek** [comment]
Can we merge this now?

**Comment by FlashTek** [comment]
@rthadur Is there something left for me to do or can this be merged into `master` now?

**Pull request merged**